src/state/database_version.rs

```diff
@@ -0,0 +1,66 @@
+use std::collections::{HashMap};
+use std::sync::Arc;
+use mdbx::{EnvironmentKind};
+use crate::kv::{MdbxWithDirHandle, tables};
+
+const DATABASE_VERSION: u64 = 1;
+
+fn init_database_version<E: EnvironmentKind>(env: &MdbxWithDirHandle<E>) -> anyhow::Result<()> {
+    let txn = env.begin_mutable()?;
+    txn.set(tables::Version, (), DATABASE_VERSION)?;
+    txn.commit()
+}
+
+fn get_datbase_version<E>(
+    env: &MdbxWithDirHandle<E>,
+) -> anyhow::Result<u64>
+    where E: EnvironmentKind,
+{
+    let txn = env.begin()?;
+    let version = match txn.get(tables::Version, ())? {
+        Some(v) => v,
+        None => 0,
+    };
+    Ok(version)
+}
+
+pub fn migrate_database<E>(
+    env: Arc<MdbxWithDirHandle<E>>,
+) where E: EnvironmentKind,
+{
+    let mut current_version;
+    if let Ok(version) = get_datbase_version(&env) {
+        current_version = version;
+    } else {
+        current_version = 0;
+    }
+
+    let migrations = HashMap::from(
+        [(0, init_database_version); 1],
```

Comment:
Please define a `type Migration = fn ... ` and replace HashMap with ordered container like BTreeMap.